core/app/components/StageSelect/StageSelectClient.tsx

```diff
@@ -24,16 +23,16 @@ export function StageSelectClient(props: Props) {
 					</div>
 					<Select
 						{...field}
-						value={props.stage?.id}
 						onValueChange={(value) => {
 							field.onChange(value);
 						}}
+						defaultValue={field.value}
 					>
-						<SelectTrigger>
-							<SelectValue placeholder="Select a stage" defaultValue={field.value}>
-								{props.stages.find((stage) => stage.id === field.value)?.name}
-							</SelectValue>
-						</SelectTrigger>
+						<FormControl>
+							<SelectTrigger>
+								<SelectValue placeholder="Select a stage" />
+							</SelectTrigger>
+						</FormControl>
```

Comment:
this seems to work better, otherwise sometimes the selector is blank after making a new selection